Automattic/jetpack-onboarding
# Fix dismissal on custom pages

It was reported that when JPO was loaded on a custom page, such as `/wp-admin/admin.php?page=jetpack`, that dismissing JPO at the end would lead to a whitescreen.

After debugging, I found that this is because we were using the `pathname` part of the URL, which only included the route parts of the URL. So, with the above url, `pathname` would be something like `/wp-admin/admin.php`. As you can see, this drops the `page` query arg, and it resulted in us sending the user to `wp-admin/admin.php`, which whitescreens.

To fix this, I just require the `url` node module that's built in and I used the `path` part of the URL.

Lastly, this PR also includes one style fix. While testing, I noticed that the dismissal button was underlined. This seemed a bit off, so I removed the underline.

To test:

- Checkout this PR
- Add the snippet below somewhere in a plugin or `functions.php`
    ```
    // add assets
    add_action( 'admin_enqueue_scripts', array( 'Jetpack_Onboarding_WelcomePanel', 'add_wizard_assets' ) );
    // add wizard HTML
    add_action( 'admin_notices', 'add_jpo_wizard' );
    function add_jpo_wizard() {
	    if ( get_option( Jetpack_Onboarding_EndPoints::HIDE_FOR_ALL_USERS_OPTION ) ) {
		    return;
	    }
    
	    ?>
	    <div id='jpo-welcome-panel'><span class='screen-reader-text'>Loading Welcome Wizard</span></div>
	    <?php
    }
    ```
- Go to the Jetpack page
- Step through the JPO setup
- Ensure you can dismiss the JPO modal and that you land back on the Jetpack page
